### PR TITLE
chore: tell `@mapeo/core` whether the app is in the foreground

### PR DESCRIPTION
**Pull request by EvanHahn**
We need to tell `@mapeo/core` whether the app is in the foreground so it can start/stop sync. This does that with a React hook.

See also: [the related `@mapeo/core` commit][0].

[0]: https://github.com/digidem/mapeo-core-next/commit/bb9b16cdf3a4eebfc1910d1c889df2b6d9afe0a5

### CODE REVIEW

File: src/frontend/useOnBackgroundedAndForegrounded.ts
Comment (ErikSin) [inline src/frontend/useOnBackgroundedAndForegrounded.ts:1]: I think this should do in the `src/frontend/hooks` folder

File: src/frontend/useOnBackgroundedAndForegrounded.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import {type MapeoClientApi} from '@mapeo/ipc';
+import {useEffect} from 'react';
+import {useAppState} from '@react-native-community/hooks';
+
+export default function useOnBackgroundedAndForegrounded(
```
Comment (ErikSin) [inline src/frontend/useOnBackgroundedAndForegrounded.ts:5]: Also, we stay away from default exports in the front end (because we are civilized people...)

**Review by ErikSin — APPROVED** [review]
Just a comment about location and use of default exports

File: src/frontend/useOnBackgroundedAndForegrounded.ts (hunk 1 above)
Comment (EvanHahn) [reply src/frontend/useOnBackgroundedAndForegrounded.ts]: 🔥 🔥 🔥

**Comment by EvanHahn** [comment]
@achou11 Yes, it's blocked on #433.

**Pull request merged**